### PR TITLE
feat(TPG >=4.23)!: Add function_docker_registry variables and use them in  terraform-google-event-function module

### PR DESCRIPTION
**Pull request by tardigrde**
This is a follow-up PR to https://github.com/terraform-google-modules/terraform-google-event-function/pull/164. It depends on that PR, so it will be in draft until the other one is merged. 

The scope of this PR is the same as the dependency PR. Variables added:
- function_docker_registry
- function_docker_repository
- function_kms_key_name

This enables storing Cloud Build container in Artifact Registry.

See https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudfunctions_function#docker_registry

### CODE REVIEW

**Comment by tardigrde** [comment]
Hi again @apeabody ! I'd then extend this module as well with the 3 variables used now in https://github.com/terraform-google-modules/terraform-google-event-function. Could you please review it + gcbrun?

**Comment by tardigrde** [comment]
Updated!

**Comment by tardigrde** [comment]
Updated the branch! @apeabody can you please move this forward somehow?

File: variables.tf [hunk 1]
```diff
@@ -122,6 +122,23 @@ variable "function_max_instances" {
...
+  type        = string
+  default     = null
+  description = "User managed repository created in Artifact Registry optionally with a customer managed encryption key. If specified, deployments will use Artifact Registry."
+}
+
+variable "function_kms_key_name" {
+  type        = string
+  default     = null
+  description = "Resource name of a KMS crypto key (managed by the user) used to encrypt/decrypt function resources."
+
```
Comment (apeabody) [inline variables.tf:141]:
```suggestion
}

```

**Comment by tardigrde** [comment]
No update since..

**Comment by apeabody** [comment]
Hi @tardigrde - From the lint it needs a `make docker_generate_docs`

```
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/README.md /tmp/tmp.h0veMwqh1U/generate_docs/workspace/README.md
73c73
< | vpc\_connector | The VPC Network Connector that this cloud function can connect to. It should be set up as fully-qualified URI. The format of this field is projects//locations//connectors/\*. | `string` | `null` | no |
---
> | vpc\_connector | The VPC Network Connector that this cloud function can connect to. It should be set up as fully-qualified URI. The format of this field is projects//locations//connectors/*. | `string` | `null` | no |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/examples/logs-slack-alerts/README.md /tmp/tmp.h0veMwqh1U/generate_docs/workspace/examples/logs-slack-alerts/README.md
30c30
< No output.
---
> No outputs.
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
```

**Comment by alexrohv** [comment]
@tardigrde @apeabody Any issues if I take over and open a new PR? I can build off the current commits. I've been waiting quite a while for this to get merged in.

**Comment by tardigrde** [comment]
@alexrohv please take over. :)

**Comment by tardigrde** [comment]
Closing since @alexrohv will take over.

Thanks so far!

**Pull request closed without merging**